.editorconfig

```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+charset = utf-8
```

Comment:
soft tabs of 4 spaces, utf-8 etc. are already the default for Rust. is there any reason to have it in the .editorconfig?

Reply:
convenient to ensure the IDE has the right settings